lib/worker/messaging.js

```diff
@@ -0,0 +1,120 @@
+const { AsyncLocalStorage } = require('async_hooks');
+function safeStringify(data) {
+  try {
+    const seen = new WeakSet();
+
+    return JSON.stringify(data, (_key, value) => {
+      if (typeof value !== "object" || value === null) {
+        return value;
+      }
+
+      if (seen.has(value)) {
+        return '[Circular]';
+      }
+
+      seen.add(value);
+      return value;
+    });
+  } catch (error) {
+    return JSON.stringify({
+      message: error.message,
+      stack: error.stack,
+    });
+  }
+}
+
+function stringify(input) {
+  if (typeof input === 'string') {
+    const start = input[0];
+    const end = input[input.length - 1];
+
+    if (
+      (start === '[' && end === ']') ||
+      (start === '{' && end === '}') ||
+      (start === '"' && end === '"')
+    ) {
+      return input;
+    }
```

Comment:
This is a super simple 'is this string json already' check, if its not, we stringify it anyway! I was not sure of a better way to do this 🤔